### ISSUE TITLE
Files performance: add test for duration testing

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/rnp#349]
## Description

The basic task of encrypting a (large) file should be not much slower than that of using just a symmetric encryption algorithm.

We should add a test that attempts round-trip encryption / decryption / signing / verification that rnp is not slower than GnuPG (we can compare and chart out).

We can generate a large file (or just download a large, static file from a fast place for reproducibility) for this purpose.

File sizes to attempt:
* 5 MB
* 50 MB
* 500 MB

**Comment by catap** [comment]
I suggest to add 1GB and 10GB files ;)

**Comment by MohitKumarAgniotri** [comment]
I would like to pick this task.

**Comment by ronaldtse** [comment]
I'd love to add 10GB but it may cause our tests to run extremely slowly :-)

**Comment by catap** [comment]
@ronaldtse well... I think we should ran it time-to-time and don't run it each push. Am I wrong?

**Comment by ronaldtse** [comment]
We can probably run the smaller ones each push and the larger ones on occasion?

@MohitKumarAgniotri and @ni4 I think this suite would be in your area too. I know @catap is interested in optimizing performance here :-)

**Comment by MohitKumarAgniotri** [comment]
I agree with @catap. we can run it periodically and keep the track of the performance results over the time.

**Comment by ni4** [comment]
@ronaldtse Should this be done on lib or CLI level? 
@MohitKumarAgniotri I partially implemented this performance check (now only for encryption, other checks should be done later) in Python. This is not merged yet, however you may check the ni4-294-cli-tests-py branch

**Comment by ronaldtse** [comment]
@ni4 I think we should do this on both levels to ensure both products we deliver (library and CLI) meet expectations. 

Hoping you could merge the said branch soon even if the tests are more basic!

**Comment by ronaldtse** [comment]
@ni4 has already provided a structure for these tests, these should be integrated into the CLI test suite merged in #294 .

**Comment by MohitKumarAgniotri** [comment]
@ni4 I am trying to run the python based test locally, but they fail. is it an expected behaviour or am I missing something?

**Comment by ni4** [comment]
@MohitKumarAgniotri Confirmed, it doesn't work well now. I need to update it to match latest changes in key/subkey generation. Will make a PR later today.
